robot.py

```diff
@@ -333,7 +359,63 @@ def teleopInit(self):
         self.drivetrain.disableVoltageCompensation()
         return
 
+    def checkTeamColorAndFieldPosition(self):
+
+        #if the bot team color selection on elastic has changed
+        selectedTeam = self.elastic.getSelectedTeam()
+        if selectedTeam != self.ELASTIC_BOT_COLOR: #checks if selection is different
+            self.ELASTIC_BOT_COLOR = selectedTeam
+            if selectedTeam == "RED":
+                teamColorIsRed = True      #will be passed to override/update function
+            elif selectedTeam == "BLUE":
+                teamColorIsRed = False
+
+            teamGyroAdjustment, teamMoveAdjustment = self.updateTeamColor(teamColorIsRed)
+            self.swervometer.updateTeam(teamGyroAdjustment, teamMoveAdjustment)
+
+        #if the bot starting position selection on elastic has changed
+        selectedPosition = self.elastic.getSelectedPosition()
+        if selectedPosition != self.ELASTIC_BOT_POSITION:
+            self.ELASTIC_BOT_POSITION = selectedPosition
+
+            positions = self.savedSwervometerConfig['ALL_ACTIVE_POSITIONS']
```

Comment:
Does this loop end up setting `fieldStartPosition` to a different value than `selectedPosition`? If not, you could call `self.updateFieldStartPosition(selectedPosition)`